biahub/track.py

```diff
@@ -0,0 +1,656 @@
+        empty_frames_idx = get_empty_frames_idx_from_csv(blank_frame_df, fov_str)
+
+    # Load label free data to check for blank frames
+    elif input_lf_dirpath is not None and blank_frame_csv_path is None:
+        click.echo(f"Loading Label Free data from: {input_lf_dirpath}...")
+        input_im_path = input_lf_dirpath / Path(*position_key)
+        im_dataset = open_ome_zarr(input_im_path)
+        channel_names = im_dataset.channel_names
+        click.echo(f"Label Free Channel names: {channel_names}")
+        for channel in input_channels["label_free"]:
+            im_arr = im_dataset[0][:, channel_names.index(channel), z_slices.start, :, :]
+            empty_frames_idx = _check_nan_n_zeros(im_arr)
+    else:
+        raise Warning(
+            "Will not check for empty frames and track can fail, please provide label free data or blank frame csv path."
+        )
+        empty_frames_idx = None
+
+    click.echo(f"Empty frames found: {empty_frames_idx}")
+
+    # Load virtual staining data, check if the required channels are present
+    if input_channels['virtual_stain'] is None:
+        raise ValueError("Virtual Staining input channels is required.")
+    # Check if the tracking channel is present in the virtual staining channels
+    if input_channels['tracking'] is None:
+        raise ValueError("Tracking input channels is required.")
+    elif len(input_channels["tracking"]) != 1:
+        raise ValueError("Only one channel is allowed and required for tracking.")
+    elif input_channels["tracking"][0] not in input_channels["virtual_stain"]:
+        raise ValueError("Tracking channel not found in Virtual Stain input channels.")
+
+    click.echo(f"Loading data from: {input_vs_path}...")
+    vs_dataset = open_ome_zarr(input_vs_path)
+    T, C, Z, Y, X = vs_dataset.data.shape
+    channel_names = vs_dataset.channel_names
+    click.echo(f"Virtual Stanining Channel names: {channel_names}")
+    yx_scale = vs_dataset.scale[-2:]
+
+    # Define output metadata
+    output_channels = []
+    for channel in input_channels["tracking"]:
+        output_channels.append(f"{channel}_labels")
+
+    output_metadata = {
+        "shape": (T, 1, 1, Y, X),
+        "chunks": None,
+        "scale": vs_dataset.scale,
+        "channel_names": output_channels,
+        "dtype": np.uint32,
+    }
+
+    create_empty_hcs_zarr(
+        store_path=output_dirpath, position_keys=[position_key], **output_metadata
+    )
+
+    # Load virtual staining data
+    for channel in input_channels["virtual_stain"]:
+        data_dict[channel] = vs_dataset[0][:, channel_names.index(channel), z_slices]
+
+    # Apply virtual staining projection function
+    if vs_projection_function is not None:
+        click.echo(
+            f"Applying {vs_projection_function.function} projection to the virtual staining data..."
+        )
+
+        projection = eval(vs_projection_function.function)  # Convert string to function
```

Comment:
Nitpicking, because avoiding `eval` is kind of a good practice.
This could be a nice alternative, assuming it's always a numpy function.
```suggestion
        projection = getattr(np, vs_projection_function.function)  # Convert string to function
```